### PR TITLE
Do not show Work Items in State Closed.

### PR DESCRIPTION
**Pull request by vikram-raj**
### [ ] What does this PR do? _[Mandatory]_
- Do not show Work Items in State Closed. 

### [ ] What issue/task does this PR references? _[Mandatory]_

https://openshift.io/openshiftio/openshiftio/plan/detail/1916

### [ ] Are the tests Included? _[Mandatory]_

Yes, #2490

### CODE REVIEW

**Comment by debloper** [comment]
Suggested an alternative way to approach this problem to Sunil, I'd suggest hold on a bit for that mockup to come out, as it may be a better way than the checkbox and situational filterbox hax.

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2505-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2505-fabric8-planner.badger.fabric8.io

File: src/app/components_ngrx/group-types-panel/group-types-panel.component.html [hunk 1]
```diff
@@ -3,9 +3,11 @@
     class="f8-group-filter__type"
     id="{{item.group}}"
     [routerLink]="[]"
-    [queryParams]="showTree ?
-      {q: fnBuildQueryParam(item), showTree: showTree} :
-      {q: fnBuildQueryParam(item)}"
+    [queryParams]="showTree && showCompleted ?
```
Comment (sanbornsen) [inline src/app/components_ngrx/group-types-panel/group-types-panel.component.html]: Why is it named ```showCompleted```, not ```showClosed```?

File: src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.html [hunk 2]
```diff
@@ -14,8 +14,10 @@
     <div class="truncate padding-right-15"
       (click)="setGuidedTypeWI(collection)">
       <a [routerLink]="[]"
-        [queryParams]="showTree ?
-          { q: constructURL(iteration.id), showTree: showTree } :
+        [queryParams]="showTree && showCompleted ?
```
Comment (sanbornsen) [inline src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.html]: This has become a bit complicated to understand. Let's create a function for this.

File: src/app/components_ngrx/group-types-panel/group-types-panel.component.html [hunk 3]
```diff
@@ -3,9 +3,11 @@
...
+      { q: fnBuildQueryParam(item), showTree: showTree , showCompleted: showCompleted } :
```
Comment (sanbornsen) [inline src/app/components_ngrx/group-types-panel/group-types-panel.component.html]: Let's create a function for this to ease of understanding.

File: src/app/components_ngrx/iterations-panel/iterations-panel.component.html [hunk 4]
```diff
@@ -32,10 +32,12 @@
           <a
           [routerLinkActive]="'f8-itr--selected'"
           [routerLink]="[]" [attr.data-id]="iteration.id"
-          [queryParams]="showTree ?
-              { q: constructURL(iteration.id), showTree: showTree } :
+          [queryParams]="showTree && showCompleted ?
+              { q: constructURL(iteration.id), showTree: showTree, showCompleted: showCompleted } :
```
Comment (sanbornsen) [inline src/app/components_ngrx/iterations-panel/iterations-panel.component.html]: Create a function for this to ease of understanding.

File: src/app/components_ngrx/group-types-panel/group-types-panel.component.html (hunk 1 above)
Comment (vikram-raj) [reply src/app/components_ngrx/group-types-panel/group-types-panel.component.html]: This comes up after Todd and Nimisha discussion on visual design. @nimishamukherjee please confirm.

**Comment by vikram-raj** [comment]
@rgarg1 Added test for this PR https://github.com/jarifibrahim/fabric8-planner/pull/6.

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2505-8 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2505-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
show completed

**Comment by nimishamukherjee** [comment]
@rgarg1 please note
> "Completed" state filtering will happen based on the selected context only

In your screenshot **vap-to be closed** does NOT match the context. If scenario is selected, the completed filter will apply to scenarios, fundamentals, and papercuts. It does not apply to the children or parent.

**Comment by rgarg1** [comment]
Thanks @nimishamukherjee for the explanation. With that limitation, the PR is working fine. Approved.

**Comment by Raunak1203** [comment]
@vikram-raj if Show completed is unchecked and we change state of a workitem from new to close... How it should behave ?
screenshot from 2018-03-23 20-58-05

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2505-9 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2505-fabric8-planner.badger.fabric8.io

**Comment by nimishamukherjee** [comment]
The comment by @Raunak1203 needs to be tracked under another issue. This is a general issue for all changes made on a work item. If we have filtered by a bunch of criteria and we make changes on a work item which is part of the result set, then the list needs to get updated.
@sahil143 @sanbornsen @vikram-raj thoughts?

**Comment by vikram-raj** [comment]
> if Show completed is unchecked and we change state of a workitem from new to close... How it should behave?

@Raunak1203 List should refresh but it is not happening right now with any attributes of the work item. We will fix this issue in separate PR.

**Comment by nimishamukherjee** [comment]
@Raunak1203 please create a separate issue for your observation but extend it to all work item edits.

**Comment by Raunak1203** [comment]
okay @nimishamukherjee @vikram-raj issue- https://github.com/openshiftio/openshift.io/issues/2823

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2505-16 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2505-fabric8-planner.badger.fabric8.io

**Pull request merged**